vcGL/src/vcTextureHelper.cpp

```diff
@@ -343,21 +343,36 @@ udResult vcTexture_SaveImage(vcTexture *pTexture, vcFramebuffer *pFramebuffer, c
   UD_ERROR_NULL(pFrameBufferPixels, udR_MemoryAllocationFailure);
 
   vcTexture_BeginReadPixels(pTexture, 0, 0, currSize.x, currSize.y, pFrameBufferPixels, pFramebuffer);
-
   UD_ERROR_CHECK(vcTexture_ConvertPixels(pFrameBufferPixels, format, &pPixelsOut, vcTextureFormat_RGBA8, currSize.x * currSize.y));
 
-  pWriteData = stbi_write_png_to_mem(pPixelsOut, 0, currSize.x, currSize.y, 4, &outLen);
-  UD_ERROR_NULL(pWriteData, udR_InternalError);
+  udSprintf(&pOutFilepath, "%s.%s", pFilename, pExtension);
+
+  if (udStrcmp(pExtension, "png") == 0)
+  {
+    int outLen = 0;
+
+    pWriteData = stbi_write_png_to_mem(pPixelsOut, 0, currSize.x, currSize.y, 4, &outLen);
+
+    UD_ERROR_NULL(pWriteData, udR_InternalError);
+    UD_ERROR_CHECK(udFile_Save(pOutFilepath, pWriteData, outLen));
+  }
+  else if (udStrcmp(pExtension, "jpg") == 0)
+  {
+    int jpgWriteResult = stbi_write_jpg(pOutFilepath, currSize.x, currSize.y, 4, pPixelsOut, 90);
```

Comment:
This will create an inconsistency between JPG and PNG as `udFile_Save` can be extended to save to more than just a file. You should change both code paths to use the `stbi_write_<ext>_to_func` functions, I'm not sure how they work, so you may not be able to use `udFile_Save` but instead the individual `udFile` functions.